pinot-segment-local/src/main/java/org/apache/pinot/segment/local/segment/readers/PinotSegmentRecordReader.java

```diff
@@ -230,6 +232,20 @@ public void getRecord(GenericRow reuse, int docId) {
     }
   }
 
+  public PrimaryKey getPrimaryKey(int docId, List<String> primaryKeyColumns) {
```

Comment:
Let's make it similar to `getRecord` by reusing the buffer primary key
```suggestion
  public void getPrimaryKey(PrimaryKey reuse, int docId, List<String> primaryKeyColumns) {
```